ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDashboard.robot

```diff
@@ -67,7 +67,11 @@ Authorize rhods-dashboard service account
 
 Login To RHODS Dashboard
    [Arguments]  ${ocp_user_name}  ${ocp_user_pw}  ${ocp_user_auth_type}
-   #Wait Until Page Contains  Log in with
+
+   # Wait until we are in the OpenShift auth page or already in Dashboard
+   ${expected_text_list} =    Create List    Log in with    Data Science Projects
```

Comment:
why `Data Science Projects`? we don't use this keyword only to login directly in DS Project page. besides, there is `Wait for RHODS Dashboard to Load` keyword which waits for the dashboard to be loaded no?

Reply:
I wanted to use a text that always appears in Dashboard, and "Data Science Projects" is always shown in the menu on the left

>Wait for RHODS Dashboard to Load keyword which waits for the dashboard to be loaded no?

This code is at the very beginning of `Login To RHODS Dashboard`. We cannot use `RHODS Dashboard to Load` yet because it is possible we'll be redirected to the login page for Dashboard

Reply:
okay but why do we need to wait at the very beginning of `Login To RHODS Dashboard` for `Data Science Projects`?

it's probably harmless, but I wonder if this wait wouldn't be covered already by `Wait for RHODS Dashboard to Load` which gets usually called after the Login keyword

Reply:
The problem I found when running the tests in local is that `Login To RHODS Dashboard` sometimes has false in `${oauth_prompt_visible} =  Is OpenShift OAuth Login Prompt Visible` because the page is still loading. 

I added this wait to avoid this error. As we could already have a session with dashboard, the wait should be checking if login is required or if we are already logged into dashboard. That's why it is waiting for "Log in with" or "Data Science Projects"

Reply:
understood, thanks